### PR TITLE
updated optional and required fields in data model

### PR DESCRIPTION
**Pull request by nithyaganeshng**
**Problem:**
Updated data model to make the fields required or optional according to specs.

**Solution:**
Used library A to complete scenario X with quality and style.

**Validation:**
Please include here how it was verified that the PR works as intended.

**Type of change:**
- [ ] Feature work
- [x] Bug fix
- [ ] Documentation
- [x] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

**Risk**:
- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [x] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [ ] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

**Work Item links**:
Please include here links for this work item, or deferred work, or related work. E.g. if the refactoring is too big to fit in this PR, or the localized strings need to be updated later, please link the TODO work items here.

**Documentation Links**:
Please include here links to any related background documentation for this PR.

### CODE REVIEW

File: sdk/src/main/java/com/microsoft/did/sdk/internal/ImageLoader.kt [hunk 1]
```diff
@@ -18,12 +18,14 @@ class ImageLoader @Inject constructor() {
 
     suspend fun loadRemoteImagesIntoContract(request: IssuanceRequest) {
         val logo = request.contract.display.card.logo
-        if (logo.image != null || logo.uri == null) {
+        if(logo != null) {
```
Comment (AndiMiko) [inline sdk/src/main/java/com/microsoft/did/sdk/internal/ImageLoader.kt:21]: oh I see the entire Logo was missing not just the url :D

File: sdk/src/main/java/com/microsoft/did/sdk/credential/service/models/oidc/PresentationRequestContent.kt [hunk 2]
```diff
@@ -17,28 +17,28 @@ import kotlinx.serialization.Serializable
...
     val redirectUrl: String = "",
 
     @SerialName("iss")
-    val issuer: String = "",
+    val issuer: String,
 
-    val scope: String = "",
+    val scope: String,
 
     val state: String = "",
```
Comment (AndiMiko) [inline sdk/src/main/java/com/microsoft/did/sdk/credential/service/models/oidc/PresentationRequestContent.kt:36]: At some point (mostly me) decided that we'd do non-nullable Strings with "" as default values. In hindsight I think that was a bad call. We may want to move to nullable Strings again in the future. (Might depend on the exact scneario though).

**Review by AndiMiko — APPROVED** [review]

File: sdk/src/main/java/com/microsoft/did/sdk/credential/service/models/oidc/PresentationRequestContent.kt (hunk 2 above)
Comment (AndiMiko) [reply sdk/src/main/java/com/microsoft/did/sdk/credential/service/models/oidc/PresentationRequestContent.kt:36]: (no action item here)

File: sdk/src/main/java/com/microsoft/did/sdk/internal/ImageLoader.kt (hunk 1 above)
Comment (nithyaganeshng) [reply sdk/src/main/java/com/microsoft/did/sdk/internal/ImageLoader.kt:21]: yeah logo is not required anymore.

File: sdk/src/main/java/com/microsoft/did/sdk/credential/service/models/oidc/PresentationRequestContent.kt (hunk 2 above)
Comment (nithyaganeshng) [reply sdk/src/main/java/com/microsoft/did/sdk/credential/service/models/oidc/PresentationRequestContent.kt:36]: I have cautiously added nullable types now where absolutely required :)

**Pull request merged**